src/Controller/BlockAdminController.php

```diff
@@ -131,31 +137,30 @@ public function switchParentAction(Request $request): Response
 
     /**
      * @throws AccessDeniedException
-     * @throws PageNotFoundException
+     * @throws BadRequestHttpException
      */
     public function composePreviewAction(Request $request): Response
     {
-        $this->admin->checkAccess('composePreview');
+        $existingObject = $this->assertObjectExists($request, true);
+        \assert(null !== $existingObject);
 
-        $blockId = $request->get('block_id');
+        $this->checkParentChildAssociation($request, $existingObject);
 
-        $block = $this->admin->getObject($blockId);
+        $this->admin->checkAccess('composePreview', $existingObject);
 
-        if (null === $block) {
```

Comment:
If we use normal id parameter, and we use `assertObjectExists` like in the actions on CRUDController, we can control by default when the provided id exists and when not.